lib/puppet/provider/scheduled_task/win32_taskscheduler.rb

```diff
@@ -401,6 +405,10 @@ def validate_trigger(value)
 
   private
 
+  def canonical_command(command)
+    File.expand_path(command).gsub(/\//, '\\')
```

Comment:
According to https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx, windows paths can contain `~`, which leads to problems like:

```ruby
irb(main):025:0> File.expand_path("C:\\Users\\Administrator").casecmp(File.expand_path("~")) == 0
=> true
```

I'd recommend we instead use `File.identical?`:

```ruby
irb(main):020:0> File.identical?("C:\\Users\\Administrator", "c:/users/administrator")
=> true
irb(main):021:0> File.identical?("C:\\Users\\Administrator", "~")
=> false
```